W-19704481-add-Agent-Fabric-authoring-content-mk (#504)
mulesoft/docs-code-builder: modules/ROOT/pages/af-agent-networks.adoc

= Building Agent Networks for Agent Fabric

> Agent Fabric manages and coordinates the execution of agents across your enterprise, regardless of where they're deployed. Agent Fabric solves the problem of integrating and automating complex processes that involve multiple agents and tools.

Reviewer: ```suggestion
Manage and coordinate the execution of agents across your enterprise, regardless of where they're deployed, with Agent Fabric. Integrate and automate complex processes that involve multiple agents and tools.
```

Author: Leaving as-is since I need to use the same content as in Agent Fabric Overview

Complex tasks often require a series of steps, with each step handled by a different agent or tool. Managing this process manually is difficult and error-prone. Agent networks (networks of agents in a domain) act as central hubs for defining, validating, and executing these multi-step processes.

> Define your agent network in a simple, human-readable YAML file in Anypoint Code Builder desktop. This approach abstracts away the underlying technical complexities and lets you focus on the business constraints and context of your process without needing to understand the inner workings of the orchestration engine.

Reviewer: ```suggestion
Define your agent network in a simple, human-readable YAML file in Anypoint Code Builder desktop. This approach abstracts away the underlying technical complexities so you focus on the business constraints and context of your process without needing to understand the inner workings of the orchestration engine.
```

Author: edited manually

> Don't worry, we give you a head start by providing you with a YAML template in your agent network project. The MuleSoft Dev Agent can help you configure it for your business environment, publish agent network assets to Anypoint Exchange, and deploy your agent network instance.

Reviewer: ```suggestion
You can find the YAML template in your agent network project. MuleSoft Dev Agent can help you configure it for your business environment, publish agent network assets to Anypoint Exchange, and deploy your agent network instance.
```

> For more information about Agent Fabric, see link:https://docs.mulesoft.com/agent-fabric/[Agent Fabric Overview].

Reviewer: Shouldn't it be xref:agent-fabric::index.adoc[]?

Author: As I mentioned in my message, inline links are still in progress. I need to verify all of them. THx

> == Considerations

Reviewer: This seems to be a "Before You Begin" topic.

> Before you start your agent network project, familiarize yourself with the Agent Fabric considerations. For more information, see link:https://docs.mulesoft.com/agent-fabric[Agent Fabric Overview].

Reviewer: (Globally) "xref" instead of "link"?

Also, make sure you understand these limitations.

* An agent network project supports only one domain (one YAML file per project). For multiple domains, create a separate project for each.
* All LLM models in your agent network project must support structured responses in JSON format. For example, a `gpt4.1` model is supported; a `gpt4` model isn't.

> * When you add governance policies to your agent network project YAML file, those policies persist between deployments. However, if you add governance policies at run-time, those policies won't persist. The next time you deploy your agent network, the governance policies used will be the ones in agent-network.yaml.

Reviewer: ```suggestion
* When you add governance policies to your agent network project YAML file, those policies persist between deployments. However, if you add governance policies at run-time, those policies don't persist. The next time you deploy your agent network, the governance policies used are the ones in the agent-network.yaml file.
```
